Model/Service/Recommendation/StateAwareCategoryService.php

```diff
@@ -289,15 +296,7 @@ private function sanitizeLimit(StoreInterface $store, $limit)
             || $limit > $maxLimit
             || $limit === 0
         ) {
-            $this->logger->debugWithSource(
-                sprintf(
-                    'Limit set to %d - original limit was %s',
-                    $maxLimit,
-                    $limit
-                ),
-                [],
-                $this
-            );
+            $this->debugWithSource(sprintf('Limit set to %d - original limit was %s', $maxLimit, $limit));
```

Comment:
This should be changed too `$this->debugWithSource('Limit set to %d - original limit was %s', $maxLimit, $limit);`. The logic for string formatting `sprintf` should be in the logger.